proxy/src/transport/tls/config.rs

```diff
@@ -63,17 +64,70 @@ pub enum Error {
     EndEntityCertIsNotValid(webpki::Error),
     InvalidPrivateKey,
     TimeConversionFailed,
+    #[cfg(target_os = "linux")]
+    InotifyInit(io::Error),
 }
 
 impl CommonSettings {
+    fn paths(&self) -> [&PathBuf; 3] {
+        [
+            &self.trust_anchors,
+            &self.end_entity_cert,
+            &self.private_key,
+        ]
+    }
+
+    /// Stream changes to the files described by this `CommonSettings`.
+    ///
+    /// The returned stream consists of each subsequent successfully loaded
+    /// `CommonSettings` after each change. If the settings could not be
+    /// reloaded (i.e., they were malformed), nothing is sent.
+    pub fn stream_changes(self, interval: Duration)
+        -> impl Stream<Item = CommonConfig, Error = ()>
+    {
+        // If we're on Linux, first atttempt to start an Inotify watch on the
+        // paths. If this fails, fall back to polling the filesystem.
+        #[cfg(target_os = "linux")]
+        // Explicit type annotation so that the compiler doesn't infer this to
+        // be `impl Stream<...>` (in which case the match arms would have
+        // different types).
```

Comment:
This comment doesn't make the code any clearer to me.

Reply:
I thought that, at first glance, the type annotation seemed unnecessary, so I felt like it ought to be documented so readers would understand why it's there?